### ISSUE TITLE
Vacuum only connects after first start

### ISSUE THREAD

**Issue by Bluetexon** [ioBroker/ioBroker.mihome-vacuum#53]
Hello,
let me explain the behavior first:
After entering the credentials for the vacuum I started an instance and it immediately started the vacuum. I think it triggered all triggers at the same time.
The main problem is now, that it works fine for some time, but when ioBroker restarts (or i restart the instance manually) the connection to the vacuum stops working. It updates the values sporadically (like every 1-5 minutes) and all the triggers stop working.
Also I have to reinstall the instance to get it working again. It works for the first install of the instance, but after that I can't seem to get it working again.
I noticed it writes this every 10 seconds in the console:
![image](https://user-images.githubusercontent.com/22718113/45436685-a517c600-b6b3-11e8-90b4-7557beafd92f.png)

Can you help me with that, please?

Thanks,
Bluetexon

**Comment by BuZZy1337** [comment]
Have you entered any things into the goto state? Or is the goto state empty?

**Comment by Bluetexon** [comment]
> Have you entered any things into the goto state? Or is the goto state empty?
> 
> Edit:
> Looks like the installation is broken.. Could you please try to reinstall the vacuum adapter?

I did a complete reinstall, but that unfortunately changed nothing. And no, I have not changed any of the states.

**Comment by BuZZy1337** [comment]
Which Node & NPM Version is running on your ioBroker host?

**Comment by Bluetexon** [comment]
I'm running Node version v8.11.4 and NPM version 6.4.1

**Comment by BuZZy1337** [comment]
Do you have any scripts running which are reading or writing anything to the states of the vacuum adapter?

**Comment by Bluetexon** [comment]
No, I don't have any scripts running which access the vacuum adapter. The version is 1.1.5

**Comment by Bluetexon** [comment]
Okay, I've got an update for that. I've took a look at the log and added this line to the main.js:
![image](https://user-images.githubusercontent.com/22718113/45489917-08623080-b766-11e8-8723-cb63c72d5555.png)
After I added that, everything seems to be working fine.

**Comment by BuZZy1337** [comment]
Yes but there must be something else wrong because the parsegoto function should not be triggered if you did not enter anything to the goto state?!

Also I am worried about the "unknown state" error message at the bottom of your Screenshot. This message is thrown if you change a state and the adapter don't know what it should do with that state.. There must be something else what is wrong on your system..

**Comment by Bluetexon** [comment]
My installation of ioBroker on my RaspberryPi was done via the ioBroker linux image and I didn't had any other problems yet. But you are right, it could be a faulty installation.
Is there any way I could check that or can I help you by sending you a log of something specific?

**Comment by BuZZy1337** [comment]
The issue was caused by the js-controller.
Has been fixed with https://github.com/ioBroker/ioBroker.js-controller/commit/3e3ae29e2bf0785b7dfa10b8eeac34984907119b and included in 1.5.2

```
cd /opt/iobroker
iobroker stop
npm i iobroker.js-controller@latest
iobroker start
```
